airflow/providers/google/cloud/transfers/bigquery_to_postgres.py

```diff
@@ -36,8 +34,6 @@ class BigQueryToPostgresOperator(BigQueryToSqlBaseOperator):
     :param postgres_conn_id: Reference to :ref:`postgres connection id <howto/connection:postgres>`.
     """
 
-    template_fields: Sequence[str] = (*BigQueryToSqlBaseOperator.template_fields, "dataset_id", "table_id")
```

Comment:
Isn't this a breaking change?

I think the real issue is with:

https://github.com/apache/airflow/blob/6802d4136fab9b8a02faf22f48cb9f6d259fd523/airflow/providers/google/cloud/transfers/bigquery_to_sql.py#L96-L99

and it will affect all operators that inhert from the base class

Reply:
> Isn't this a breaking change?
> 
> I think the real issue is with:
> 
> https://github.com/apache/airflow/blob/6802d4136fab9b8a02faf22f48cb9f6d259fd523/airflow/providers/google/cloud/transfers/bigquery_to_sql.py#L96-L99
> 
> and it will affect all operators that inhert from the base class

Coming to think of it, it might be breaking indeed as fields that don't exist in the parent's `template_fields' are removed from child's definition.
I suggest reverting it for now.
@romsharon98 instead of deleting this line, try to hardcode all of the values that should be templated, and see if it works (a bit ugly, but I don't have better idea for now).

Reply:
Thanks for your insight @eladkal !
Can you help me understand why is this breaking change?

This is how I understand it:
Lets assume I revert the PR, so both `"dataset_id", "table_id"` are templated field for `BigQueryToPostgresOperator`.

But the parent constructor (`BigQueryToSqlBaseOperator`) always run them over by the line you mentioned.

So as I understand it this reverted line has no meaning.

Reply:
@eladkal is right - but we shiould not revert it - instead we should ad those to fields to the base class.

Reply:
> @eladkal is right - but we shiould not revert it - instead we should ad those to fields to the base class.

Sounds good to me, a note
From a technical perspective of the validation pre-commit - 
As the validation is currently based on very simplified AST parsing, it would be better for now to define the fields directly (i.e., `template_fields = ['a','b']`), rather than relying on parents' fields (i.e, `template_fields=(**ParentClass.template_fields,'b')`)., otherwise the validation might fail.
The cost would be minimal abuse to the inheritance, which can later be fixed.

Reply:
Correct. This is really what I also proposed - to move template_fields = ['dataset_id', 'template_id'] to `BigQueryToSqlBaseOperator`. This is where they belong and this is what will make them consistent with the AST check.

Reply:
Followup PR https://github.com/apache/airflow/pull/36663